File: backend/src/main/java/com/conlaboro/entity/Favorite.java
package com.conlaboro.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.OffsetDateTime;

@Data
@TableName("favorites")
public class Favorite {

    @TableId(type = IdType.AUTO)
    private Long id;
    private Long userId;
    private Long projectId;
    @TableField("created_at")
    private OffsetDateTime createdAt;
}

File: frontend/src/api/favorite.js
import request from './request'

/** 收藏 / 取消收藏（切换） */
export function toggleFavorite(projectId) {
  return request.put(`/favorites/${projectId}`)
}

/** 检查是否已收藏 */
export function isFavorited(projectId) {
  return request.get(`/favorites/${projectId}`)
}
